### ISSUE TITLE
add loss functions

### ISSUE THREAD

**Issue by c-schicho** [EnablingIntelligence/NeuralNetworksFromScratch#7]
add the following loss functions:
- mse
- l1
- l2
- cross-entropy
